src/main/java/com/parkingapp/parkingservice/infrastructure/database/JdbcVehiclesRepository.java

```diff
@@ -0,0 +1,66 @@
+package com.parkingapp.parkingservice.infrastructure.database;
+
+import com.parkingapp.parkingservice.domain.vehicle.Vehicle;
+import com.parkingapp.parkingservice.domain.vehicle.VehicleRepository;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public class JdbcVehiclesRepository implements VehicleRepository {
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+    public JdbcVehiclesRepository(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
+        this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
+    }
+
+    @Override
+    public boolean saveVehicle(Vehicle vehicle) {
+        MapSqlParameterSource params = new MapSqlParameterSource()
+                .addValue("id", vehicle.getId())
+                .addValue("brand", vehicle.getBrand())
+                .addValue("model", vehicle.getModel())
+                .addValue("color", vehicle.getColor())
+                .addValue("plate", vehicle.getPlate())
+                .addValue("country", vehicle.getCountry())
+                .addValue("userId", vehicle.getUserId());
+
+        return namedParameterJdbcTemplate.update(
+                """
+                INSERT INTO vehicles(id, brand, model, color, plate, country, user_id)
+                VALUES (:id, :brand, :model, :color, :plate, :country, :userId)
+                ON CONFLICT(plate, user_id) DO NOTHING;
+                """,
+                params
+        ) > 0;
+    }
+
+    @Override
+    public Optional<Vehicle> getVehicleByUserIdAndPlate(UUID userId, String plate) {
+        String sql = """
+                SELECT * FROM vehicles
+                WHERE user_id = :userId AND plate = :plate
+                """;
+
+        SqlParameterSource params = new MapSqlParameterSource()
+                .addValue("userId", userId)
+                .addValue("plate", plate);
+
+        RowMapper<Vehicle> rowMapper = (rs, rowNum) -> new Vehicle(
```

Comment:
Extract mapper outside of method to be able to re use it by another method in the future

Reply:
👍🏻